src/map/utils/fishingcontest.h

```diff
@@ -0,0 +1,150 @@
+﻿/*
+===========================================================================
+
+  Copyright (c) 2010-2015 Darkstar Dev Teams
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see http://www.gnu.org/licenses/
+
+===========================================================================
+*/
+
+#ifndef _FISHINGCONTEST_H
+#define _FISHINGCONTEST_H
+
+#include "../items/item_fish.h"
+#include "../utils/fishingutils.h"
+
+#include <cstring>
+#include <list>
+#include <map>
+#include <vector>
+
+class CBasicPacket;
+class CCharEntity;
+
+enum FISHING_CONTEST_CRITERIA : uint8
+{
+    CONTEST_CRITERIA_SIZE   = 0,
+    CONTEST_CRITERIA_WEIGHT = 1,
+    CONTEST_CRITERIA_BOTH   = 2
+};
+
+enum FISHING_CONTEST_MEASURE : uint8
+{
+    CONTEST_MEASURE_GREATEST = 0,
+    CONTEST_MEASURE_SMALLEST = 1
+};
+
+enum FISHING_CONTEST_STATUS : uint8
+{
+    CONTEST_STATUS_CONTESTING = 0,
+    CONTEST_STATUS_OPENING    = 1,
+    CONTEST_STATUS_ACCEPTING  = 2,
+    CONTEST_STATUS_RELEASING  = 3,
+    CONTEST_STATUS_PRESENTING = 4,
+    CONTEST_STATUS_HIATUS     = 5,
+    CONTEST_STATUS_CLOSED     = 6,
+};
+
+struct FishingContest
+{
+    FISHING_CONTEST_STATUS   status     = CONTEST_STATUS_CONTESTING;
+    FISHING_CONTEST_CRITERIA criteria   = CONTEST_CRITERIA_SIZE;
+    FISHING_CONTEST_MEASURE  measure    = CONTEST_MEASURE_GREATEST;
+    uint16                   fishId     = 0;
+    uint32                   startTime  = 0xEFFFFFFF; // to allow for changetimes to not overflow
```

Comment:
Can you explain this? It isn't immediately obvious how this works

Reply:
I can probably get rid of this and just use 0 as the start time -- When I initially coded this on Horizon, I was storing completed contests in the database and starting new ones in cache before updating the DB, so I needed a start time trigger of '0' to flag the contest as beginning, and I used a guaranteed future time just to show that it's in a creation state.  I didn't use 0xFFFFFFFF because if it somehow got through and calculated stage timings, it would overflow.